### DISCUSSION TITLE
useWatch with array of field names does not coerce field types

### DISCUSSION THREAD

**Discussion by jonotrujillo** [react-hook-form/react-hook-form#4431] [Q&A]
Hello everyone 👋 

When using the `useWatch` hook with an array of fields, field values are not coerced to their types. For example, given that a field is of type `Date`, `useWatch` will return it as an object, with all optional attributes.

See an example here: https://codesandbox.io/s/wonderful-lumiere-4w28t?file=/src/App.tsx

In this example, it is problematic to use some of the Date methods, as this error is returned:

```
This expression is not callable.
  Type '{}' has no call signatures.
```

Is this behaviour intended? What would be the best practice to get the correct type?

**Comment by bluebill1049** [comment]
good catch, fixing that in v7.
